### ISSUE TITLE
windows pkg.remove error

### ISSUE THREAD

**Issue by jfindlay** [saltstack/salt#22298]
### version
```yaml
jmoney-windows-12:
               Salt: 2015.2.0rc2
             Python: 2.7.8 (default, Jun 30 2014, 16:08:48) [MSC v.1500 64 bit (AMD64)]
             Jinja2: 2.7.3
           M2Crypto: 0.21.1
     msgpack-python: 0.4.5
       msgpack-pure: Not Installed
           pycrypto: 2.6
            libnacl: Not Installed
             PyYAML: 3.11
              ioflo: Not Installed
              PyZMQ: 14.5.0
               RAET: Not Installed
                ZMQ: 4.0.5
               Mako: Not Installed
```
### master config
```yaml
win_repo: '/srv/salt/win/repo'
win_repo_mastercachefile: '/srv/salt/win/repo/winrepo.p'
win_gitrepos:
  - 'https://github.com/saltstack/salt-winrepo.git'
```

### setup
```yaml
# salt-run winrepo.genrepo
# salt-run winrepo.update_git_repos
# salt windows-minion pkg.refresh_db
# salt jmoney-windows-12 pkg.install Firefox
jmoney-windows-12:
    ----------
    Firefox:
        ----------
        new:
            35.0.1
        old:
    Mozilla Maintenance Service:
        ----------
        new:
            35.0.1
        old:
```
### execution
```yaml
# salt jmoney-windows-12 pkg.remove Firefox
jmoney-windows-12:
    ERROR: Unable to run command ['C:\\Program Files \\Mozilla Firefox\\uninstall\\helper.exe', '/S'] with the context {'with_communicate': True, 'shell': False, 'env': {'TMP': 'C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\2', 'COMPUTERNAME': 'CLOUD-SERVER-51', 'USERDOMAIN': 'CLOUD-SERVER-51', 'PSMODULEPATH': 'C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules\\', 'COMMONPROGRAMFILES': 'C:\\Program Files\\Common Files', 'PROCESSOR_IDENTIFIER': 'Intel64 Family 6 Model 45 Stepping 7, GenuineIntel', 'PROGRAMFILES': 'C:\\Program Files', 'PROCESSOR_REVISION': '2d07', 'SYSTEMROOT': 'C:\\Windows', 'PATH': 'C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\', 'PROGRAMFILES(X86)': 'C:\\Program Files (x86)', 'TEMP': 'C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\2', 'COMMONPROGRAMFILES(X86)': 'C:\\Program Files (x86)\\Common Files', 'PROCESSOR_ARCHITECTURE': 'AMD64', 'ALLUSERSPROFILE': 'C:\\ProgramData', 'LOCALAPPDATA': 'C:\\Users\\Administrator\\AppData\\Local', 'HOMEPATH': '\\Users\\Administrator', 'USERDOMAIN_ROAMINGPROFILE': 'CLOUD-SERVER-51', 'PROGRAMW6432': 'C:\\Program Files', 'USERNAME': 'Administrator', 'LOGONSERVER': '\\\\CLOUD-SERVER-51', 'PROMPT': '$P$G', 'COMSPEC': 'C:\\Windows\\system32\\cmd.exe', 'PROGRAMDATA': 'C:\\ProgramData', 'SESSIONNAME': 'RDP-Tcp#0', 'PATHEXT': '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC', 'CLIENTNAME': 'poseidon', 'FP_NO_HOST_CHECK': 'NO', 'WINDIR': 'C:\\Windows', 'HOMEDRIVE': 'C:', 'APPDATA': 'C:\\Users\\Administrator\\AppData\\Roaming', 'SYSTEMDRIVE': 'C:', 'NUMBER_OF_PROCESSORS': '2', 'PROCESSOR_LEVEL': '6', 'COMMONPROGRAMW6432': 'C:\\Program Files\\Common Files', 'OS': 'Windows_NT', 'PUBLIC': 'C:\\Users\\Public', 'USERPROFILE': 'C:\\Users\\Administrator'}, 'stdout': -1, 'stdin': None, 'stderr': -2, 'cwd': 'C:\\Users\\Administrator'}, reason: [Error 2] The system cannot find the file specified
```

**Comment by jfindlay** [comment]
@twangboy

**Comment by twangboy** [comment]
I've only seen this on 32bit systems. The problem is that the win_repo sls files don't get parsed the same way state files do. They are raw yaml. So you can't add logic to account for 32bit and 64bit paths.

The path under your execution portion above looks strange though. It's misssing the (x86) after Program Files...

**Comment by jfindlay** [comment]
I encountered this while running manual release tests and the system was a 64 bit system.  I was able to remove `adobereader` without problem, so this might be something funny with the `Firefox` package.

**Comment by jfindlay** [comment]
Possibly related to #22039.

**Comment by UtahDave** [comment]
OK, this is fixed. I can successfully remove Firefox now.

I'm going to open up another issue, though, because if Firefox is already uninstalled, then we get this same error. I think we need to have salt check for the existence of the uninstaller before attempting to execute it.

**Comment by twangboy** [comment]
Fixed by #24980
Will be available in 2015.8 (Beryllium)
